### ISSUE TITLE
Rendering issues with calendar events

### ISSUE THREAD

**Issue by bandtank** [espocrm/espocrm#2861]
**Describe the bug**
When an event is created (meeting, call, etc.) with times that span the boundary of a day (rolling from one day into the next), the event is visually displayed as an all-day event even if that isn't true. The event should be shown on both days at the relevant times (see below).

**To Reproduce (MANDATORY, DO NOT REMOVE)**
Explicit steps to reproduce the behavior:
1. Set the application's timezone to America/Denver and Date Format to YYYY-MM-DD
2. Go to the calendar
3. Set the calendar to Week mode by clicking the button at the top right between Month and Timeline (the user's calendar, not a shared calendar)
4. Create an event:
  * Type: Call
  * Name: Test
  * Start: 2023-09-18 at 00:00
  * End: 2023-09-19 at 00:30 (30-minute duration)
It should look like this, which is correct:
![image](https://github.com/espocrm/espocrm/assets/1170415/47e14da7-9ed4-4764-becb-6d45935b8bf3)

5. Edit the event:
  * Start: 2023-09-17 at 23:30
  * End: 2023-09-18 at 00:30 (1-hour duration)
It looks like this, __which is not correct__ (the event should appear at the bottom of 2023-09-17 and top of 2023-09-18):
![image](https://github.com/espocrm/espocrm/assets/1170415/0813f7d8-bd25-4ab8-85f5-18de5f4fd261)

6. Edit the event:
  * Start: 2023-09-17 at 23:30
  * End: 2023-09-18 at 01:30 (2-hour duration)
It looks like this, __which is not correct in a different way__ (the event should appear at the bottom of 2023-09-17 and top of 2023-09-18):
![image](https://github.com/espocrm/espocrm/assets/1170415/759f1c09-cb12-43c0-99d2-58993f152135)

7. Edit the event:
  * Start: 2023-09-17 at 01:30
  * End: 2023-09-17 at 03:30 (2-hour duration)
It looks like this, __which is not correct in a different way again__ (the event should appear between 01:30 and 03:30, but it starts at 00:00 and ends at 01:00, which is also the wrong duration):
![image](https://github.com/espocrm/espocrm/assets/1170415/b4d7b1fc-2c1a-439b-a129-3619d8fdba0e)

8. Edit the event:
  * Start: 2023-09-17 at 03:30
  * End: 2023-09-17 at 05:30 (2-hour duration)
It looks like this, __which is finally correct__:
![image](https://github.com/espocrm/espocrm/assets/1170415/7371c4e7-71f8-435d-9e02-6e0fa089631e)

9. Edit the event:
  * Start: 2023-09-17 at 03:30
  * End: 2023-09-18 at 05:30 (1-day, 2-hour duration)
 It looks like this, __which is not correct__ (the event should partially span both days):
![image](https://github.com/espocrm/espocrm/assets/1170415/44e9e2de-8463-4fa2-8274-81d473a05e11)
Using the left and right arrows to scroll between weeks results in the event looking like this upon scrolling back to the target week (scroll left by 1 week and then right by 1 week to get back to the original week):
![image](https://github.com/espocrm/espocrm/assets/1170415/a52bf6e8-1942-43c7-a832-0b14e555d93c)

**Expected behavior**
* Calendar events should partially span days when events are not marked as All-Day.
* Calendar events should be displayed at the correct times according to the selected timezone.
* Calendar events should visually render correctly and consistently.

**Screenshots**
This is how a similar event looks in Outlook. Notice the event spans each day partially:
![image](https://github.com/espocrm/espocrm/assets/1170415/b9f99f00-4e43-42eb-bb76-cabdc0c113b8)

**EspoCRM version**
The issue has been primarily observed and tested in 8.0.2, but several of the behaviors are also evident in 7.5.6.

**Additional context**
This bug is present in a new instance of EspoCRM as well as in an instance with extensions and customizations. The issue can also be seen in Day mode, but it more obvious in Week mode (agendaWeek at the very least, but maybe in basicWeek as well).

**Comment by yurikuzn** [comment]
I don't consider this as a bug. I remember I considered this behavior as acceptable from the beginning.

**Comment by yurikuzn** [comment]
> It looks like this, which is not correct in a different way again (the event should appear between 01:30 and 03:30, but it starts at 00:00 and ends at 01:00, which is also the wrong duration):

**Comment by bandtank** [comment]
Unfortunately, the calendar shows the wrong times in many cases. I became aware of this after my users complained about events not appearing at the right times. One of my examples shows a case of an event not appearing at the right times even without a multi-day period. I think that is a significant bug.

**Comment by yurikuzn** [comment]
As I understood, there are 3 separate issues. Multi-day events I consider as a feature request. Maybe for the future minor/major release.

2 others I need yet to reproduce.

**Comment by yurikuzn** [comment]
Reproduced one issue: https://github.com/espocrm/espocrm/issues/2863.

**Comment by bandtank** [comment]
I will work on reproducing the other issues with more details. When I have more information, I will post it here.

**Comment by yurikuzn** [comment]
As I understand the issue with wrong time happend only after edit. When page is refreshed everything is correct.

**Comment by bandtank** [comment]
I think that is true. My previous observation was that the event stayed at the wrong time on the calendar, but I could have done something incorrectly. All recent attempts at reproducing the issue with an event being listed at the wrong time on a single day seem to be fixed by refreshing as you suggested.

**Comment by yurikuzn** [comment]
Fixed: https://github.com/espocrm/espocrm/commit/fe77f2c14ca722f5f2a1df9eab882a7aca4a5382

**Comment by yurikuzn** [comment]
Another issue: https://github.com/espocrm/espocrm/issues/2865
